packages/api/app/index.js

```diff
@@ -13,10 +13,35 @@ const awsServerlessExpressMiddleware = require('aws-serverless-express/middlewar
 
 const { services } = require('@cumulus/aws-client');
 const { MissingRequiredEnvVar } = require('@cumulus/errors');
+const { secretsManager } = require('@cumulus/aws-client/services');
+const Logger = require('@cumulus/logger');
 
 const router = require('./routes');
 const { jsonBodyParser } = require('./middleware');
 
+const log = new Logger({ sender: '@api/index' });
+
+// Load Environment Variables
+// This should be done outside of the handler to minimize Secrets Manager calls.
+const initEnvVarsFunction = async () => {
+  if (!process.env.api_config_secret_id) {
+    log.warn('No process.env.api_config_secret_id set, *not* attempting to provide environment variable defaults');
+    return;
+  }
+  const response = await secretsManager().getSecretValue(
+    { SecretId: process.env.api_config_secret_id }
+  ).promise();
+  if (response.SecretString === undefined) {
+    throw new Error(`AWS Secret did not contain a stored value: ${process.env.api_config_secret_id}`);
+  }
+  const envSecret = JSON.parse(response.SecretString);
+  log.warn('Setting env vars');
+  log.warn(envSecret);
+  process.env = { ...envSecret, ...process.env };
```

Comment:
```suggestion
  process.env = { ...process.env, ...envSecret };
```

I assume we want anything that comes out of our secret to override existing environment variables on `process.env`?

Reply:
It's worth a discussion, but this was intentional to allow lambda.env to override the deployment secret.    This allows for manual intervention/reconfiguration on the fly *per lambda* and/or testing applications.

Reply:
Ah I see. I could imagine leaving it as is then